### ISSUE TITLE
When `--color-only` is used, colors don't extend to the whole line

### ISSUE THREAD

**Issue by da-x** [dandavison/delta#136]
When using `--color-only`, the color only ends at the last line's non-white-space character, instead of extending to fill also the blank space after to the last character to the right side of the terminal.

Compare:

![image](https://user-images.githubusercontent.com/321273/80455868-398c7600-8935-11ea-9df5-a0fb5c2adf4b.png)

With:

![image](https://user-images.githubusercontent.com/321273/80455910-4f01a000-8935-11ea-9068-5e7386934152.png)

p.s. I'm author of [da-x/fancydiff](https://github.com/da-x/fancydiff). I think `delta` is very cool, and I'd longed for abandoning `fancydiff` for a better option.

**Comment by dandavison** [comment]
Hi @da-x, fancydiff looks cool! It's a bit more ambitious than delta in some ways -- I'm very happy that I've got syntect and the sublime syntax definitions. If you ever feel like writing git diff processing code in Rust instead of Haskell, we have some open issues here :)

 So, `--color-only` is working as intended I believe. It's job is to "not alter the input in any way other than applying colors", so that includes not right-padding with spaces . AFAIK the only way that delta can apply a background color to the right-hand edge of the terminal is by right-padding with space characters and applying the background color to the spaces.

On the plus side, `--color-only` is equivalent to

```
--keep-plus-minus-markers --width variable --tabs 0 --commit-style plain \
--file-style plain --hunk-style plain
```

so you should be able to achieve the effect you want by using some subset of those options directly and not using `--color-only`. (It's `--width variable` that you want to get rid of.)

**Comment by da-x** [comment]
@dandavison I'm quite positive that `fancydiff` achieves this without space padding. I'll check how

**Comment by da-x** [comment]
`echo '\x1b[48;2;40;100;200mTest\x1b[K'`

Yeah, it's the `\x1b[K` ANSI control code.

**Comment by da-x** [comment]
Ah, it's not enough. I also pass the following environment to `less`: `LESSANSIENDCHARS=mK`.

**Comment by dandavison** [comment]
Wow, thanks! I had never looked into it. This seems to be potentially a big advance. It might be perceptibly more efficient for the terminal emulator than having to handle a ton of space characters, and it may simplify some of the terminal-width accounting code. By default, delta spawns a child `less` process, so we should be able to set that env var. As far as you know this is well supported in terminal emulators?

Here's a screenshot showing some playing around with this (iTerm2):

<table><tr><td><img width=600px src="https://user-images.githubusercontent.com/52205/80494305-86238180-8934-11ea-9de3-e4ab42deedea.png" alt="image" /></td></tr></table>

@lzybkr should this all work fine with (your) `less.exe` on Windows and common Windows terminal emulators?

**Comment by lzybkr** [comment]
@dandavison - I think `\x1b[K` should work fine with Windows 10 terminal emulators and with my fixes to `less.exe`.

**Comment by dandavison** [comment]
Thanks very much @da-x and @lzybkr. The master branch now uses `\x1b[K` and sets `LESSANSIENDCHARS` in the child `less` process.

**Comment by dandavison** [comment]
Good catch! I believe that's fixed in master via #167.

(I noticed another similar bug in that I was no longer terminating with a reset sequence; it didn't affect the behavior of my terminal emulator but I believe it is correct to reset at the end of each line, it is what `ansi_term::ANSIString` does, and it did cause a problem when using Delta with Magit in Emacs)

**Comment by dandavison** [comment]
(I've been bad about unit tests for ANSI escape sequences -- that is needed.)
